Foundation/URLSession/URLSession.swift

```diff
@@ -419,6 +424,22 @@ open class URLSession : NSObject {
     open func dataTask(with url: URL) -> URLSessionDataTask {
         return dataTask(with: _Request(url), behaviour: .callDelegate)
     }
+
+    /*
+     * data task convenience methods.  These methods create tasks that
+     * bypass the normal delegate calls for response and data delivery,
+     * and provide a simple cancelable asynchronous interface to receiving
+     * data.  Errors will be returned in the NSURLErrorDomain,
+     * see <Foundation/NSURLError.h>.  The delegate, if any, will still be
+     * called for authentication challenges.
+     */
+    open func dataTask(with request: URLRequest, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
+        return dataTask(with: _Request(request), behaviour: .dataCompletionHandler(completionHandler))
+    }
+
+    open func dataTask(with url: URL, completionHandler: @escaping (Data?, URLResponse?, Error?) -> Void) -> URLSessionDataTask {
+        return dataTask(with: _Request(url), behaviour: .dataCompletionHandler(completionHandler))
+    }
```

Comment:
Moved these from an extension to the class itself. With Darwin Foundation we're able to override these methods in subclasses.

Reply:
SE-0117 strikes again.

I think you can mark _each_ method in the extension with `open`, and that should reduce the size of this PR and keep changes easier to see.

Reply:
They were already `open`, it’s just that we can’t override methods declared in extensions.

Reply:
Sorry, I didn't double checked my proposal.

I just tested in a small Xcode (Mac) project that reproduces the problem, and I get this error "Overriding non-`@objc` declarations from extensions is not supported". I tried adding a simple `@objc` in front of the extension in the Swift framework, and it seems to fix the problem.

Do you think that might be a better solution? I don't mind moving the methods into the main class code, but if someone wrote them as an extension for organization, I would prefer to keep them that one.

Reply:
I’m afraid `@objc` won’t work on non-Darwin platforms.

Reply:
That's true. What a strange limitation of extensions. I wonder why `open` there doesn't generate a warning, if it cannot be overriden anyway.

Reply:
AFAIK this has to do with method dispatch. We can't retroactively add methods to class vtable, unless we use some completely different dispatching mechanism, like message passing in Objective-C. That's why `@objc` works — it relies on Objective-C runtime.